### ISSUE TITLE
SSH to the cluster failed to start.

### ISSUE THREAD

**Issue by jeffcruise** [telepresenceio/telepresence#1489]
### What were you trying to do?

I was trying to swap deployment on my local kind cluster. This was my first time using telepresence.

### What did you expect to happen?

I expected a terminal to show up, but instead it just complained about ssh.

### What happened instead?

(please tell us - the traceback is automatically included, see below.
use https://gist.github.com to pass along full telepresence.log)

### Automatically included information

Command line: `['/usr/local/bin/telepresence', '--swap-deployment', 'cluster-operators-controller-manager']`
Version: `0.108-13-gf692885`
Python version: `3.7.6 (default, Feb 24 2020, 15:17:32) 
[GCC 7.4.0]`
kubectl version: `Client Version: v1.16.15 // Server Version: v1.16.15`
oc version: `(error: [Errno 2] No such file or directory: 'oc': 'oc')`
OS: `Linux PC0YTPGX-LT 4.15.0-48-generic #51-Ubuntu SMP Wed Apr 3 08:28:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`

```
Traceback (most recent call last):
  File "/usr/local/bin/telepresence/telepresence/cli.py", line 135, in crash_reporting
    yield
  File "/usr/local/bin/telepresence/telepresence/main.py", line 68, in main
    socks_port, ssh = do_connect(runner, remote_info)
  File "/usr/local/bin/telepresence/telepresence/connect/connect.py", line 119, in do_connect
    args.from_pod
  File "/usr/local/bin/telepresence/telepresence/connect/connect.py", line 70, in connect
    raise RuntimeError("SSH to the cluster failed to start. See logfile.")
RuntimeError: SSH to the cluster failed to start. See logfile.

```

Logs:

```
32 | Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
  39.5  32 |                  node.kubernetes.io/unreachable:NoExecute for 300s
  39.5  32 | Events:
  39.5  32 |   Type    Reason     Age        From                                     Message
  39.5  32 |   ----    ------     ----       ----                                     -------
  39.5  32 |   Normal  Scheduled  <unknown>  default-scheduler                        Successfully assigned cluster-operators-system/cluster-operators-d58fd2d7565045a79f693919f24641ff to cluster-operator-control-plane
  39.5  32 |   Normal  Pulled     32s        kubelet, cluster-operator-control-plane  Container image "datawire/telepresence-k8s:0.108" already present on machine
  39.5  32 |   Normal  Created    32s        kubelet, cluster-operator-control-plane  Created container manager
  39.5  32 |   Normal  Started    31s        kubelet, cluster-operator-control-plane  Started container manager
  39.5 TEL | [32] ran in 0.17 secs.

```

**Comment by jeffcruise** [comment]
My problem was when my deployment specified a `securityContext.runAsUser` in my deployment spec. It specified a UID of 9999. This caused sshd to fail when trying to log someone in.

```
          securityContext:
            runAsUser: 9999                                                                                                                                                                                                                              
            allowPrivilegeEscalation: false
```

I was able to diagnose the issue by running telepresence with swap-deployment, then exec'ing into the resulting pod. I killed the existing `sshd` in the container and ran it myself with `/usr/sbin/sshd -d`, which foregrounds the process and outputs logs to stderr. Maybe this troubleshooting step will be useful for others.

The fix was to remove my `runAsUser` configuration from the deployment spec I was trying to replace. This might not be suitable for every use-case, so IMO it warrants a fix in telepresence itself.

I'm a bit busy right now, but I can post repro steps a little later.
